### PR TITLE
FIX-#2776: Use best practices when starting Ray

### LINKED ISSUE #2776: Use best practices in initializing Ray with plasma
ray-project/ray#14326

### PR DESCRIPTION
**Pull request by devin-petersohn**
Resolves #2776

Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2776 
- [ ] tests added and passing

### CODE REVIEW

**Comment by devin-petersohn** [comment]
cc @rkooo567

File: docs/out_of_core.rst [hunk 1]
```diff
@@ -1,46 +1,28 @@
...
-Out of core is detected from an environment variable set in bash.
-
-.. code-block:: bash
-
-   export MODIN_OUT_OF_CORE=true
-
-We also set up a way to tell Modin how much memory you'd like to use. Currently, this
-only accepts the number of bytes. This can only exceed your memory if you have enabled
-``MODIN_OUT_OF_CORE``.
+Out of core is now enabled by default for both Ray and Dask engines.
```
Comment (vnlitvinov) [inline docs/out_of_core.rst:11]: Should we add version notes for Ray and Dask here? Should we bump the requirements?

**Review by vnlitvinov — APPROVED** [review]

File: docs/out_of_core.rst (hunk 1 above)
Comment (vnlitvinov) [reply docs/out_of_core.rst:11]: I've also noticed that it _might_ start working starting with Ray 1.3.0 which isn't yet released.
Am I being wrong here?

**Comment by vnlitvinov** [comment]
@devin-petersohn based on https://docs.ray.io/en/master/memory-management.html?highlight=object%20spilling#object-spilling and https://github.com/modin-project/modin/issues/2776#issuecomment-795915058, maybe we should postpone this PR until Ray 1.3.0 is released and we have moved to it successfully?..

cc @YarShev

**Comment by devin-petersohn** [comment]
@vnlitvinov Yes, that is the plan to postpone until Ray 1.3.

**Comment by devin-petersohn** [comment]
I wanted to get something out for people who are currently operating on Ray nightly wheels. A couple of people have requested it.

**Comment by vnlitvinov** [comment]
> @vnlitvinov Yes, that is the plan to postpone until Ray 1.3.

I suggest to mark this as draft then.

**Comment by YarShev** [comment]
@devin-petersohn , are the PR and the issue up-to date? Should we close those?

**Comment by devin-petersohn** [comment]
@YarShev relevant parts of the PR were merged with #3132

**Comment by YarShev** [comment]
@devin-petersohn , what about the issue?

**Pull request closed without merging**